### DISCUSSION TITLE
When do new versions of periodic job schedules take effect?

### DISCUSSION THREAD

**Discussion by tony-pizza** [sidekiq/sidekiq#6424] [Q&A] [answered]
I want to better understand the versioning behavior of periodic job schedules. I haven't found much documentation about versioning in the wiki but I've built my own understanding through manual testing and looking at the source. Could you confirm my understanding below is correct?

- Versions are based on all periodic job data, so the version only changes when job data changes (schedule, args, job options). (This one's pretty well-documented in source code, but sharing for completeness / in case it's helpful for others.)
- Versions are installed / take effect only when a Sidekiq process gains leadership
- The version that is live will always correspond to the schedule in the leader process's application code. (This is obvious perhaps, but feels worth noting that Sidekiq has no awareness of the "latest" version. For example, if an "old" Sidekiq process keeps or gains leadership during a blue/green deploy, its schedule will remain the current periodic jobs version for as long as it remains the leader, even if new Sidekiq processes with an updated periodic jobs schedule have started.)

Thank you!

**Comment by mperham** [comment, marked as answer]
That’s all exactly right. Sidekiq Enterprise makes no attempt to “order” the versions and keep track of the latest. The last leader determines the current periodic data.
